src/reducer.js

```diff
@@ -142,16 +152,30 @@ const RegistrationLiteReducer = (state = DEFAULT_STATE, action) => {
         case PAY_RESERVATION: {
             return { ...state, checkout: payload.response, reservation: null, userProfile: null, invitation: null };
         }
-        case GET_MY_INVITATION:{
-            return {...state, invitation: payload.response};
+        case GET_MY_INVITATION: {
+            return { ...state, invitation: payload.response };
         }
-        case CLEAR_MY_INVITATION:{
-            return {...state, invitation: null};
+        case CLEAR_MY_INVITATION: {
+            return { ...state, invitation: null };
         }
         case UPDATE_CLOCK: {
             const { timestamp } = payload;
             return { ...state, nowUtc: timestamp };
         }
+        case REQUEST_TICKET_DISCOUNT: {
+            const { promoCode } = payload;
+            return { ...state, promoCode: { ...state.promoCode, code: promoCode, loading: true } }
+        }
+        case REMOVE_TICKET_DISCOUNT: {
+            return { ...state, promoCode: { loading: true } }
+        }
+        case GET_TICKET_DISCOUNT: {
```

Comment:
this action is not needed u need to refactor REQUESTED_TICKET_TYPES to set the current promo code ( if its passed by param)